### PR TITLE
Compatibility regarding nftables

### PR DESCRIPTION
**Pull request by papanito**
Fixes #2319

### CODE REVIEW

**Comment by papanito** [comment]
@JoshVanL did you already had a chance to look at my PR?

**Comment by papanito** [comment]
Hi @JoshVanL, @munnerz this pr is open since quite a while. Do you mind check it?

**Review by wallrj — APPROVED** [review]
Thanks for this @papanito 

And sorry for the not replying sooner.

There are some merge conflicts right now. 
Will you have some time to rebase the branch?
If not, I'll do it and hopefully we can get this merged finally.

**Comment by papanito** [comment]
@wallrj can you have a look?

File: content/en/docs/installation/compatibility.md [hunk 1]
```diff
@@ -5,9 +5,18 @@ weight: 100
...
 Below you will find details on various compatibility issues and quirks that you
 may be effected by in your environment.
 
+- [GKE](#gke)
+- [AWS EKS](#aws-eks)
+- [Webhook](#webhook)
+- [iptables vs. nftables](#iptables-vs-nftables)
+  - [Switch to `iptables`](#switch-to-iptables)
+  - [Configure network provider](#configure-network-provider)
+    - [Calico](#calico)
```
Comment (wallrj) [inline content/en/docs/installation/compatibility.md:18]: Please remove this extra ToC.
There is already a [document ToC on the right](https://deploy-preview-187--cert-manager.netlify.app/docs/installation/compatibility/) which is maintained automatically.

File: content/en/docs/installation/compatibility.md [hunk 2]
```diff
@@ -43,6 +51,37 @@ deployment, or, if using Helm, configuring it in your `values.yaml` file.
...
 `webhook.securePort` value must be changed to a different, free port.
 
-
 ## Webhook
+
 Disabling the webhook is not supported anymore since `v0.14`.
+
+## iptables vs. nftables
+
+Some distros - like [Debian 10](https://wiki.debian.org/nftables) or [RHEL 8](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.0_release_notes/rhel-8_0_0_release#networking) - recently switched from iptables to [nftables](https://wiki.nftables.org/) which causes some unexpected behaviour:
```
Comment (wallrj) [inline content/en/docs/installation/compatibility.md:60]:
```suggestion
Some distributions - like [Debian 10](https://wiki.debian.org/nftables) or [RHEL 8](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.0_release_notes/rhel-8_0_0_release#networking) - recently switched from iptables to [nftables](https://wiki.nftables.org/) which causes some unexpected behaviour:
```

**Review by wallrj — CHANGES_REQUESTED** [review]
Thanks @papanito 

I forgot to add the `/ok-to-test` tag in the last review, and now I see that [there are some spellcheck errors](https://prow.build-infra.jetstack.net/view/gcs/jetstack-logs/pr-logs/pull/cert-manager_website/187/pull-cert-manager-website-verify/1340960400572682240) ....which are difficult to see because the CI output isn't syntax highlighted.

You'll need to add `iptables`, `nftables`, `distros` `RHEL` etc to the `.spelling` file in the repo root.

You can also run `make verify` locally to get clearer spell check output.

(ignore all the grammar and other suggestions that it spits out).

File: content/en/docs/installation/compatibility.md (hunk 1 above)
Comment (papanito) [reply content/en/docs/installation/compatibility.md]: Removed the toc, please have a look again

**Comment by papanito** [comment]
Just read your comment, will do the missing stuff as well

**Comment by wallrj** [comment]
Hey @papanito 

Thanks for the updates. I discussed this with the team at standup this morning and it was mentioned that we should link to the Kubernetes website documentation on  nftables compatibility ...but when I went looking I found that they've removed it from the doc: https://github.com/kubernetes/website/pull/19773 because kube-proxy  / kubeadm now support nftables (since Kubernetes 1.17)...algthough they didn't remove  the documentation until 1.18.

So perhaps we should simply link to the original Kubernetes docs for 1.17 and note that this workaround is only necessary on K8S < 1.17?
 * https://v1-17.docs.kubernetes.io/docs/setup/production-environment/tools/kubeadm/#ensure-iptables-tooling-does-not-use-the-nftables-backend

The docs for 1.17 add one extra command than in the 1.16 docs:
 * https://v1-16.docs.kubernetes.io/docs/setup/production-environment/tools/kubeadm/#ensure-iptables-tooling-does-not-use-the-nftables-backend
```
# ensure legacy binaries are installed
sudo apt-get install -y iptables arptables ebtables
...
```

Anyway, the consensus is that we should try to avoid duplicating workarounds that are already in the Kubernetes docs  because ours will likely get out of date. Let me know what you think?

**Comment by papanito** [comment]
> should try to avoid duplicating workarounds that are already in the Kubernetes docs

Fully agree. I will give it a try and improve my changes...

**Comment by wallrj** [comment]
Hey @papanito I'll close this PR now, but if you have time, please followup with a shorter PR which notes the nftables problem and which versions of Kubernetes are affected, with a link to the K8S documented workaround for those versions.

Thanks.

**Pull request closed without merging**